engine_test.go

```diff
@@ -1286,3 +1322,30 @@ func newCtx() *sql.Context {
 	pid++
 	return sql.NewContext(context.Background(), sql.WithPid(pid))
 }
+
+type lockableTable struct {
+	sql.Table
+	readLocks  int
+	writeLocks int
+	unlocks    int
+}
+
+func newLockableTable(t sql.Table) *lockableTable {
+	return &lockableTable{Table: t}
+}
+
+var _ sql.Lockable = (*lockableTable)(nil)
+
+func (l *lockableTable) Lock(ctx *sql.Context, write bool) error {
+	if write {
+		l.writeLocks++
```

Comment:
Why do we count locks?
Isn't it 1/0 (SET/UNSET) for the same session?

Reply:
...or is it  just counter for number of read/writes for unlock?

Reply:
is just a counter for the test